solr/core/src/java/org/apache/solr/update/processor/NestedUpdateProcessorFactory.java

```diff
@@ -72,8 +72,8 @@ public void processAdd(AddUpdateCommand cmd) throws IOException {
   }
 
   private void processDocChildren(SolrInputDocument doc, String fullPath) {
-    int childNum = 0;
     for(SolrInputField field: doc.values()) {
+      int childNum = 0;
```

Comment:
If the test did not show this bug, lets tweak the test to show it.  Yeah?

Reply:
I'll add a test with another key holding childDocs to ensure this bug does not resurface